File: android/app/src/main/java/com/example/shepherdproduct/layer_data/entity/SearchEntity.kt
package com.example.shepherdproduct.layer_data.entity

import kotlinx.serialization.Serializable

@Serializable
data class SearchEntity (
    val header: Header,
    val body: Body
)

data class Body (
    val pageNo: Long,
    val totalCount: Long,
    val numOfRows: Long,
    val items: MutableList<Item>
)

data class Item (
    val ENTRPS: String,
    val PRDUCT: String,
    val DSPS_DT: String? = null
)

data class Header (
    val resultCode: String,
    val resultMsg: String
)

File: ios/ShepherdProduct/SearchEntity.swift
//
//  SearchEntity.swift
//  ShepherdProduct
//
//  Created by k4keye on 2022/03/09.
//

import Foundation

struct SearchEntity: Codable{
    let header : Header
    let body : Body
    
}

struct Header : Codable{
    let resultCode: String
    let resultMsg: String
}

struct Body : Codable{
    let pageNo: Int
    let totalCount: Int
    let numOfRows: Int
    let items: [Item]
}

struct Item : Codable{
    let ENTRPS: String
    let PRDUCT: String
    let DSPS_DT: String?
}


struct SearchResult {
    var code: Bool
    let items: [Item]?
}
